test/framework/filetools.py

```diff
@@ -494,6 +494,53 @@ def test_guess_patch_level(self):
         ]:
             self.assertEqual(ft.guess_patch_level([patched_file], self.test_buildpath), correct_patch_level)
 
+    def test_back_up_file(self):
+        """Test back_up_file function."""
+        fh, fp = tempfile.mkstemp()
+        os.close(fh)
+
+        ft.write_file(fp, 'foobar')
```

Comment:
define a variable for the file contents rather than copy-pasting `'foobar'` everywhere?